js/src/dom/selectorEngine.js

```diff
@@ -50,12 +48,71 @@ const SelectorEngine = (() => {
     }
   }
 
+  const scopeSelectorRegex = /:scope\b/
+  const supportScopeQuery = (() => {
+    const element = document.createElement('div')
+    try {
+      element.querySelectorAll(':scope *')
+    } catch (e) {
+      return false
+    }
+
+    return true
+  })()
+
+  let findFn = null
+  if (supportScopeQuery) {
+    findFn = Element.prototype.querySelectorAll
+  } else {
+    findFn = function (selector) {
+      if (!scopeSelectorRegex.test(selector)) {
+        return this.querySelectorAll(selector)
+      }
+
+      const hasId = Boolean(this.id)
+      if (!hasId) {
+        // eslint-disable-next-line no-magic-numbers
+        this.id = `q${Math.floor(Math.random() * 9000000) + 1000000}`
```

Comment:
Use `Util.getUID` (https://github.com/twbs/bootstrap/blob/v4-without-jquery/js/src/util.js#L51) instead of that